ms2deepscore/MS2DeepScore.py

```diff
@@ -0,0 +1,125 @@
+from typing import List
+from typing import Union
+import numpy as np
+from matchms.similarity.BaseSimilarity import BaseSimilarity
+from tqdm import tqdm
+
+from ms2deepscore.vector_operations import cosine_similarity
+from ms2deepscore.vector_operations import cosine_similarity_matrix
+from ms2deepscore import BinnedSpectrum
+from ms2deepscore import SpectrumBinner
+from ms2deepscore.models import SiameseModel
+
+
+class MS2DeepScore(BaseSimilarity):
+    """Calculate MS2DeepScore similarity scores between a reference and a query.
+
+    Using a trained model, binned spectrums will be converted into spectrum
+    vectors using a deep neural network. The MS2DeepScore similarity is then
+    the cosine similarity score between two spectrum vectors.
+
+    Example code to calcualte MS2DeepScore similarities between query and reference
+    spectrums:
+
+    .. code-block:: python
+
+        from matchms import calculate_scores()
+        from matchms.importing import load_from_json
+        from ms2deepscore import MS2DeepScore
+        from ms2deepscore.models import load_ms2ds_model
+
+        references = load_from_json("abc.json")
+        queries = load_from_json("xyz.json")
+
+        model = ... TODO: implement such a function
+        binned_references = model.spectrum_binner.transform(references)
+        binned_queries = model.spectrum_binner.transform(queries)
+
+        similarity_measure = MS2DeepScore(model)
+        scores = calculate_scores(binned_references, binned_queries)
```

Comment:
Good point. I missed one part. But I would still keep the `calculate_scores` function since that will return a matchms.Scores object rather than a numpy array (and is more in line with the matchms workflow).